docs/source/conf.py

```diff
@@ -430,6 +430,30 @@ def _check_object_validity(obj):
                 ''.format(obj, repr(doc)))
 
 
+def _get_sourcefile_and_linenumber(obj):
+    # Retrieve the original function wrapped by contextlib.contextmanager
+    if callable(obj):
```

Comment:
Do we need to check if `obj` is actually created by `contextmanager`?

Reply:
Hmm, as far as I know, `__closure__` is best we can do to determine that.

Reply:
OK